### PR TITLE
New Holodeck Programs

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds new holodeck programs:
* Teahouse
image

* Kitchen
image

* Clownworld (secure)
image

* Asylum (secure)
image

## Why It's Good For The Game

Adds functionality and more gimmicks to the holodeck.

## Changelog
:cl:
add: Added new holodeck programs.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Dont include any items that can be interacted with to make another item: Flashes can have wire cutters used on them to extract a free flashbulbs and the trees should be that special type that cant be broken.

**Comment by eeSPee** [comment]
> Dont include any items that can be interacted with to make another item

Noted.

> trees should be that special type that cant be broken.

Already covered in the holodeck code. Holo-trees and machines drop no loot.

**Comment by NickEmbased** [comment]
Looks fun

File: code/modules/holodeck/holodeck_map_templates.dm [hunk 1]
```diff
@@ -113,6 +113,18 @@
 	description = "benis"
 	mappath = "_maps/templates/holodeck_skatepark.dmm"
 
+/datum/map_template/holodeck/teahouse
+	name = "Holodeck - Japanese Tea House"
+	template_id = "holodeck_teahouse"
+	description = "benis"
```
Comment (bloons3) [inline code/modules/holodeck/holodeck_map_templates.dm:119]: really?

File: code/modules/holodeck/holodeck_map_templates.dm [hunk 2]
```diff
@@ -113,6 +113,18 @@
...
+	mappath = "_maps/templates/holodeck_teahouse.dmm"
+
+/datum/map_template/holodeck/kitchen
+	name = "Holodeck - Holo-Kitchen"
+	template_id = "holodeck_kitchen"
+	description = "benis"
```
Comment (bloons3) [inline code/modules/holodeck/holodeck_map_templates.dm:125]: again

File: code/modules/holodeck/holodeck_map_templates.dm [hunk 3]
```diff
@@ -163,3 +175,17 @@
 	description = "benis"
 	mappath = "_maps/templates/holodeck_refuelingstation.dmm"
 	restricted = TRUE
+
+/datum/map_template/holodeck/asylum
+	name = "Holodeck - Asylum"
+	template_id = "holodeck_asylum"
+	description = "benis"
```
Comment (bloons3) [inline code/modules/holodeck/holodeck_map_templates.dm:182]: cmon now

File: code/modules/holodeck/holodeck_map_templates.dm [hunk 4]
```diff
@@ -163,3 +175,17 @@
...
+	mappath = "_maps/templates/holodeck_asylum.dmm"
+	restricted = TRUE
+
+/datum/map_template/holodeck/clownworld
+	name = "Holodeck - Clown World"
+	template_id = "holodeck_clownworld"
+	description = "benis"
```
Comment (bloons3) [inline code/modules/holodeck/holodeck_map_templates.dm:189]: change

**Review by bloons3 — CHANGES_REQUESTED** [review]

**Comment by bloons3** [comment]
Why add a holodeck full of clown simplemobs?

File: code/modules/holodeck/holodeck_map_templates.dm (hunk 1 above)
Comment (eeSPee) [reply code/modules/holodeck/holodeck_map_templates.dm:119]: Description is 'benis' on all programs.

File: code/modules/holodeck/holodeck_map_templates.dm (hunk 4 above)
Comment (eeSPee) [reply code/modules/holodeck/holodeck_map_templates.dm:189]: I don't even know why it's there. It is never used/refferenced... Ever.

**Comment by eeSPee** [comment]
Description is 'benis' on all programs. I don't know why it's even there. It's never used or refferenced.

It's addressed in #4467.

**Review by utilbird — APPROVED** [review]
🅱️

File: code/modules/holodeck/holo_effect.dm [hunk 5]
```diff
@@ -109,11 +109,29 @@
...
+	mobtype = /mob/living/simple_animal/butterfly
+
+/obj/effect/holodeck_effect/mobspawner/clown
+	mobtype = list (/mob/living/simple_animal/hostile/retaliate/clown = 10,
+	/mob/living/simple_animal/hostile/retaliate/clown/banana = 6, /mob/living/simple_animal/hostile/retaliate/clown/honkling = 6,
+	/mob/living/simple_animal/hostile/retaliate/clown/fleshclown = 3, /mob/living/simple_animal/hostile/retaliate/clown/longface = 3,
+	/mob/living/simple_animal/hostile/retaliate/clown/mutant = 1, /mob/living/simple_animal/hostile/retaliate/clown/mutant/blob = 1)
+
+/obj/effect/holodeck_effect/mobspawner/psycho
+	mobtype = list (/mob/living/simple_animal/hostile/psycho/regular = 9,
```
Comment (bloons3) [inline code/modules/holodeck/holo_effect.dm:134]: A holodeck program is loading in **hostile** mobs from "TheFactory", an away mission which was removed for being extremely broken?
image

Comment (eeSPee) [reply code/modules/holodeck/holo_effect.dm:134]: Yea, I chose them cause they're gimmicky and look nice. They don't deal enviro damage and won't do a lot of damage when set loose. Keep in mind this is a restricted program, available with emag only, and is not as dangerous as beephobia or plasma flood.

Comment (bloons3) [reply code/modules/holodeck/holo_effect.dm:134]: Silicons can enable restricted programs as well…

**Comment by OceanFish1** [comment]
Oh wow, someone's using Factory mobs
Cheers, man

**Pull request merged**